netbsdsrc/share/man/man7/release.7
.\"	$NetBSD: release.7,v 1.1.2.1 1997/11/25 05:29:16 thorpej Exp $
.\"
.\" Copyright (c) 1997 The NetBSD Foundation, Inc.
.\" All rights reserved.
.\"
.\" Authors: Charles M. Hannum and Jason R. Thorpe.
.\"
.\" Redistribution and use in source and binary forms, with or without
.\" modification, are permitted provided that the following conditions
.\" are met:
.\" 1. Redistributions of source code must retain the above copyright
.\"    notice, this list of conditions and the following disclaimer.
.\" 2. Redistributions in binary form must reproduce the above copyright
.\"    notice, this list of conditions and the following disclaimer in the
.\"    documentation and/or other materials provided with the distribution.
.\" 3. All advertising materials mentioning features or use of this software
.\"    must display the following acknowledgement:
.\"        This product includes software developed by the NetBSD
.\"        Foundation, Inc. and its contributors.
.\" 4. Neither the name of The NetBSD Foundation nor the names of its
.\"    contributors may be used to endorse or promote products derived
.\"    from this software without specific prior written permission.
.\"
.\" THIS SOFTWARE IS PROVIDED BY THE NETBSD FOUNDATION, INC. AND CONTRIBUTORS
.\" ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
.\" TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
.\" PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE FOUNDATION OR CONTRIBUTORS
.\" BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
.\" CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
.\" SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
.\" INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
.\" CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
.\" ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
.\" POSSIBILITY OF SUCH DAMAGE.
.\"
.Dd November 18, 1997
.Dt RELEASE 7
.Os NetBSD
.Sh NAME
.Nm release
.Nd layout of NetBSD releases and snapshots
.Sh DESCRIPTION
This document describes the layout of
.Nx
releases and snapshots.
This layout should be consistent between FTP servers and CD-ROMs,
except possibly the path that leads to the release hierarchy.
.Pp
In this document, the following special words have these definitions:
.Bl -tag -width "<machine_arch>"
.It Em <machine>
The platform for which the release was built, corresponding to the
.Em hw.machine
sysctl variable, e.g.
.Em i386
or
.Em amiga .
.It Em <machine_arch>
The architecture for which a particular installation set was built,
corresponding to the
.Em hw.machine_arch
sysctl variable, e.g.
.Em i386
or
.Em m68k .
.It Em <rel>
The target release.
.El
.Pp
All
.Sy README
files are descriptions of the various files in directories that have
.Dq non-standard
contents.
.Pp
All
.Sy BSDSUM
files are historic BSD checksums for the various files in that directory,
in the format produced by the command:
.Sy cksum -o 1 <file> .
.Pp
All
.Sy CKSUM
files are POSIX checksums for the various files in that directory, in the
format produced by the command:
.Sy cksum <file> .
.Pp
All
.Sy MD5
files are MD5 digests for the various files in that directory, in the
format produced by the command:
.Sy cksum -m <file> .
.Pp
All
.Sy SYSVSUM
files are historic AT&T System V UNIX checksums for the various files in
that directory, in the format produced by the command:
.Sy cksum -o 2 <file> .
.Pp
The MD5 digest is the safest checksum, followed by the POSIX checksum.
The other two checksums are provided only to ensure that the widest possible
range of system can check the integrity of the release files.
.Pp
Files that end in
.Sy .tgz
are gzipped tar archives.  This is used in lieu of
.Sy .tar.gz
because the software used to download the sets may incorrectly auto-unpack
files ending in
.Sy .gz
and to accomodate systems which only support 3 character extensions
to file names.
.Pp
All tar archives are relative to the target's
.Sy /
directory, and
.Em do not
include the leading
.Dq / .
.Pp
All compression of release files is to be performed with the command:
.Sy gzip -9 .
.Pp
The root of the release hierarchy is
.Sm off
.Xo
.Sy .../NetBSD-
.Em <rel>
.Sy /
.Em <machine>
.Sy / .
.Xc
.Sm on
For files which may be shared by all systems of the same
.Em <machine_arch> ,
(e.g. the
.Sy games
and
.Sy xbase
sets for m68k-based platforms), this hierarchy will have a second root of
.Sm off
.Xo
.Sy .../NetBSD-
.Em <rel>
.Sy /
.Em <machine_arch>
.Sy /
.Xc
.Sm on
which will contain those files.  For files which are completely
machine-independent (e.g. the
.Sy xfont
set), this hierarchy will have a third root of
.Sy .../NetBSD- Ns Em <rel> Ns Sy /mi/
which will contain those files.
Unless otherwise noted, all files may be symbolic links to common
.Em <machine_arch>
or machine-independent files located in their respective directory
hierarchies.
.Bl -tag -width "installation/"
.It Sy SOURCE_DATE
a file containing the date, in UTC, of the source code from which the
release or snapshot was built, in the default format produced by the
command:
.Sy date -u
.It Sy INSTALL
Installation notes, including complete descriptions of files contained
within the release hierarchy
.It Sy binary/
system binaries
.Bl -tag -width "security/"
.It Sy sets/
installation sets
.Bl -tag -width "xcontrib.tgz"
.It Sy BSDSUM
.It Sy CKSUM
.It Sy MD5
.It Sy SYSVSUM
.It Sy Split/
installation sets split with the
.Xr split 1
command.
.Em "Note: split installation sets may not be available for all platforms" .
.Bl -tag -width "base.aa"
.It Sy BSDSUM
.It Sy CKSUM
.It Sy MD5
.It Sy SYSVSUM
.It Sy base.aa
.It Sy base.ab
.It Sy ...
.El
.It Sy base.tgz
The base binary distribution.  This set contains the base
.Nx
utilities that are necessary for the system to run and be minimally
functional.  It includes shared libraries for those architectures that
support support them.  This set excludes all things listed in the sets
described below.
.It Sy comp.tgz
The compiler tools distribution.  This set contains the C and C++
compilers, assembler, linker, other toolchain components, and their
manual pages.  It also includes the system include files
.Pq Pa /usr/include
, and the static system libraries.
.It Sy etc.tgz
This set contains the system configuration files that reside in
.Pa /etc
and in several other places throughout the file system hierarchy.
.It Sy games.tgz
This set includes the games and their manual pages.
.It Sy kern.tgz
This set includes a generic kernel.
.It Sy man.tgz
This set includes all of the manual pages for the binaries and other
software contained in the
.Sy base
set which are not included in the other sets.
.It Sy misc.tgz
This set includes the system dictionaries (which are rather large), the
typesettable document set, and manual pages for other architectures, which
happen to be installed from the source tree by default.
.It Sy text.tgz
This set includes the
.Nx
text processing tools, including
.Xr groff 1 ,
all related programs, and their manual pages.
.It Sy xbase.tgz
This set includes the base X11 distribution, including manual pages and
shared libraries for those architectures that support them, and excluding
everything contained in the other X11 sets.
.It Sy xcomp.tgz
This set includes the X11 include files and static X11 libraries.
.It Sy xcontrib.tgz
This set includes binaries and manual pages for programs built from the
X11
.Dq contrib
sources.
.It Sy xfont.tgz
This set includes the X11 fonts.
.It Sy xserver.tgz
This set includes the X servers and manual pages for <machine>.
.Em "Note: this set may not be available on some platforms" .
.El
.It Sy kernel/
suitably named, gzipped kernels
.Bl -tag -width "netbsd-GENERIC.gz"
.It Sy BSDSUM
.It Sy CKSUM
.It Sy MD5
.It Sy README
.It Sy SYSVSUM
.It Sy netbsd-GENERIC.gz
A kernel built from the
.Sy GENERIC
kernel configuration file.  This is meant as an example only; different
platforms may have differently named kernels.
.El
.It Sy security/
security-related system binaries
.Bl -tag -width "secr.tgz"
.It Sy BSDSUM
.It Sy CKSUM
.It Sy MD5
.It Sy SYSVSUM
.It Sy Split/
.Bl -tag -width "secr.aa"
.It Sy BSDSUM
.It Sy CKSUM
.It Sy MD5
.It Sy SYSVSUM
.It Sy secr.aa
.It Sy secr.ab
.It Sy ...
.El
.It Sy secr.tgz
This set contains executables which are built from the
.Dq Pa src/domestic
portion of the
.Nx
source tree.  It can only be found on those sites
which carry the complete
.Nx
distribution and can legally obtain it.
(Remember, because of United States law, it may not be legal to distribute
this set to locations outside the United States and Canada.)
.El
.El
.It Sy installation/
installation helper items
.Bl -tag -width "diskimage/"
.It Sy diskimage/
disk images, for those platforms that provide them
.Bl -tag -width "diskimage-rz25.gz"
.It Sy BSDSUM
.It Sy CKSUM
.It Sy MD5
.It Sy README
.It Sy SYSVSUM
.It Sy diskimage-rz25.gz
.El
.It Sy floppy/
floppy images, for those platforms that provide them
.Bl -tag -width "floppy-144.gz"
.It Sy BSDSUM
.It Sy CKSUM
.It Sy MD5
.It Sy README
.It Sy SYSVSUM
.It Sy floppy-144.gz
.El
.It Sy miniroot/
miniroot images, for those platforms that provide them
.Bl -tag -width "miniroot.gz"
.It Sy BSDSUM
.It Sy CKSUM
.It Sy MD5
.It Sy README
.It Sy SYSVSUM
.It Sy miniroot.gz
.El
.It Sy misc/
miscellaneous installation helper utilities, including boot selectors,
floppy writing software, other software that runs under foreign operating
systems, etc.
.Bl -tag -width "CKSUMS"
.It Sy BSDSUM
.It Sy CKSUM
.It Sy MD5
.It Sy README
.It Sy SYSVSUM
.It Sy ...
.El
.It Sy netboot/
network boot programs
.Bl -tag -width "netboot.gz"
.It Sy BSDSUM
.It Sy CKSUM
.It Sy MD5
.It Sy README
.It Sy SYSVSUM
.It Sy netboot.gz
.El
.It Sy tapeimage/
tape images, for those platforms that provide them
.Bl -tag -width "tapeimage-hp9144.gz"
.It Sy BSDSUM
.It Sy CKSUM
.It Sy MD5
.It Sy README
.It Sy SYSVSUM
.It Sy tapeimage-hp9144.gz
.El
.El
.El
.Sh SEE ALSO
.Xr cksum 1 ,
.Xr date 1 ,
.Xr gzip 1 ,
.Xr split 1 ,
.Xr tar 1
.Sh HISTORY
The
.Nm
manual page first appeared in
.Nx 1.3 .


ace/ChangeLog
Fri Oct 05 05:56:50 2001  Balachandran  <bala@cs.wustl.edu>

        * ACE version 5.2 released.

Thu Oct  4 17:38:21 2001  Steve Huston  <shuston@riverace.com>

        * ace/Synch.h: Added some comments to explain the rationale for
          the different return values from the blocking and timed
          versions of ACE_Null_Semaphore::acquire().  Thanks to Doug
          Schmidt for explaining this.

Thu Oct  4 17:27:39 2001  Steve Huston  <shuston@riverace.com>

        * examples/C++NPv1/Logging_Handler.cpp (recv_log_record): Insure that
          the payload message block is properly aligned for any CDR
          extraction. Many thanks to Bala for noticing this problem and
          sending code for the fix.

Thu Oct  4 17:08:39 2001  Steve Huston  <shuston@riverace.com>

        * examples/C++NPv1/Process_Per_Connection_Logging_Server.cpp:
          In prepare(), call ACE_Process_Options::command_line() rather
          than ACE_Process_Options::process_name() to set the program
          path for the child process.

Thu Oct  4 13:18:37 2001  Steve Huston  <shuston@riverace.com>

        * examples/C++NPv1/Logging_Handler.cpp: Use a ACE_Message_Block,
          rather than hijacking a CDR block, to read the payload into.

Thu Oct  4 10:14:34 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* THANKS: Added Donald Acton to the Hall of Fame.

Thu Oct  4 06:39:20 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* ace/config-sunos5.5.h: Added ACE_LACKS_MKSTEMP as SunOS 5.5 does
	  not support mkstemp (). 
	
	* ace/config-sunos5.6.h: Undefined ACE_LACKS_MKSTEMP as Sun 5.6
	  and above support it. 

	  Thanks to Don Hinton <dhinton@swan.fr> for reporting this
	  problem and Chad Elliott for testing this out on a Sun OS 5.5
	  box. 

Wed Oct  3 16:37:41 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* ace/CDR_Stream.cpp (read_1): Reverted this change "Wed Oct  3
	  15:05:43 2001  Jeff Parsons <parsons@cs.wustl.edu>".  This
	  change is alone not enough to fix the problem that Brad
	  reported. The sample program for the text needs to be fixed to
	  get it working right.

Wed Oct  3 15:05:43 2001  Jeff Parsons <parsons@cs.wustl.edu>

        * ace/CDR_Stream.cpp (read_1);

          Reverted changed made in 

          Fri Aug 17 16:56:16 2001  Jeff Parsons <parsons@cs.wustl.edu>

          to read_1(). A similar change made in several other methods
          is ok, but in this case bytes can be copied into the
          input CDR stream without changing wr_ptr, so the check will
          fail unnecessarily. Thanks to Brad Hoskins <BHoskins@slo.newport.com>
          for pointing this out.

Wed Oct  2 12:18:37 2001  Steve Huston  <shuston@riverace.com>

        * examples/C++NPv1/Logging_Server.cpp: Fixed spelling error.
        * examples/C++NPv1/Logging_Handler.cpp: Comment length adjustment.

Tue Oct  2 21:52:31 2001  Steve Huston  <shuston@riverace.com>

        * examples/C++NPv1/Logging_Client.cpp: Reverted this:
          Mon Oct  1 18:27:41 2001  Steve Huston  <shuston@riverace.com>

Tue Oct  2 19:52:57 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* Makefile: Removed some of the files that no longer exist in the
	  repository. The inclusion of these files in the Makefile added
	  noise during the beta cutting process.

Tue Oct 02 15:34:49 2001  Carlos O'Ryan  <coryan@uci.edu>

	* etc/tao_cosevent.doxygen:
	* etc/tao_cosnotification.doxygen:
	  Fixed typos in the tag file names.

Tue Oct  2 15:52:27 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* THANKS: Added Sunny Leung to the Hall of Fame.

Tue Oct  2 15:33:36 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* THANKS: Removed one instance of Peter Phillips
	  <pphillip@opentext.com> from the Hall of Fame.

Tue Oct  2 15:00:37 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* ace/CDR_Stream.cpp (clone_from): Copied the GIOP major and minor
	  versions from the incoming CDR stream. Thanks to Sunny Leung 
	  <sleung@opentext.com> for reporting this. This closes bugzilla
	  #1043.  

Tue Oct  2 09:45:39 2001  Steve Huston  <shuston@riverace.com>

        * tests/libDLL_Test.icc:
        * tests/libService_Config_DLL.icc: Add link(exportAll) to the link
          options to export the entrypoints needed for the tests.

Mon Oct  1 20:04:14 2001  Priyanka Gontla  <pgontla@ece.uci.edu>

	* THANKS: 
	  Added Tushar Nair to the THANKS file.

Mon Oct  1 18:06:48 2001  Priyanka Gontla  <pgontla@ece.uci.edu>

	* ace/Object_Manager.cpp (ACE_UnhandledExceptionFilter):
	  Removed the un-needed #if 0's. Left them before as a safety
	  measure (in case something doesnt work) and forgot to remove
	  them later.

	* ace/Log_Msg.cpp (instance):
	  Fixed the seg faults that are raised when ace is compiled with
	  ACE_NTRACE=0 on SunOS. Added a check before locking and
	  releasing a lock.

	  Thanks to Tushar Nair <ntushar@yahoo.com> for reporting the
	  problem and for providing a fix.

Mon Oct  1 18:27:41 2001  Steve Huston  <shuston@riverace.com>

        * examples/C++NPv1/Logging_Client.cpp: Remove the third (optional)
          arg from getline(). Using a char literal without further
          qualification causes confusion in Unicode builds.

Mon Oct  1 14:44:13 2001  Ossama Othman  <ossama@uci.edu>

	* ace/SSL/SSL_Context.cpp (ssl_library_init):

	  Initialize the locking callbacks in the crypto library before
	  initializing the OpenSSL library.  OpenSSL may require the
	  locking callbacks to be initialized before its own
	  initialization.

	* ace/SSL/SSL_Context.cpp (ssl_library_fini):

	  Perform crypto library resource cleanup before reclaiming the
	  locks used by the locking callbacks.  Crypto library resource
	  cleanup makes use of the locking callbacks.

Mon Oct  1 15:19:11 2001  Douglas C. Schmidt  <schmidt@macarena.cs.wustl.edu>


ace/tao/release.pl
# release.pl,v 1.10 1998/03/18 13:18:00 levine Exp
#
# Takes over some of the release work rather than having it all
# inline in the Makefile.

$date = `/usr/bin/date +"%a %b %d %T %Y"`;
chop $date;
$VERSION = "VERSION";
$CHANGELOG = "ChangeLog";
$TAO_VERSION = '';
$MODNAME = 'TAO';

# Range of values on this is "alpha", "beta", or "major",
# in which case the 3rd, 2nd, or 1st digit.
$BUMP_WHICH = $ENV{'RELEASE_TYPE'};

sub inplace {
    my($ext, @files) = @_;
    my(@nfiles) = ();
    my($nfile);
    $ext = '~' if ($ext eq '');
    foreach $file (@files) {
        $nfile = $file . $ext;
        if (rename ($file, $nfile)) {
            push(@nfiles, $nfile);
        }
        else {
            warn "Unable to rename $files[$i] for in-place editing: $!\n";
        }
    }
    @nfiles;
}

@inplace = inplace('.del', $VERSION, $CHANGELOG);

open (VERSION, $inplace[0])
    || die "Unable to open file $inplace[0] for inplace edit: $!\n";
open (VERSIONOUT, ">$VERSION")
    || die "Unable to open file $VERSION for writing: $!\n";

undef $version_number;
while (<VERSION>) {
    if ($BUMP_WHICH =~ m/major/i) {
        $bump_expr = 'sprintf("$1%d.0.0", $2+1)';
    }
    elsif ($BUMP_WHICH =~ m/beta/i) {
        $bump_expr = 'sprintf("$1$2.%d.0", $3+1)';
    }
    else {
        $bump_expr = 'sprintf("$1$2.$3.%d", $4+1)';
    }
    $subst = 's/(TAO version )(\d+)\.(\d+)\.(\d+)/' . $bump_expr . '/e';
    eval $subst;
    ($version_number = $_) =~ s/.*(\d+\.\d+\.\d+).*/$1/ if (!defined($version_number));
    if (s/(, released ).*/$1$date./) {
        ($TAO_VERSION = $_) =~ s/^This is //;
    }
    print VERSIONOUT $_;
}

close (VERSIONOUT);
close (VERSION);
chmod 0644, $VERSION;

$message = $date."  ".$ENV{"SIGNATURE"}."  <".$ENV{"LOGNAME"}."\@cs.wustl.edu>\n\n\t* ".$TAO_VERSION."\n";
$message_printed = 0;

open (CHANGELOG, $inplace[1])
    || die "Unable to open file $inplace[1] for inplace edit: $!\n";
open (CHANGELOGOUT, ">$CHANGELOG")
    || die "Unable to open file $CHANGELOG for writing: $!\n";

while (<CHANGELOG>) {
    print CHANGELOGOUT "$message" if ( ! $message_printed++ );
    print CHANGELOGOUT $_;
}

close (CHANGELOG);
close (CHANGELOGOUT);
chmod 0644, $CHANGELOG;

# Morph the version number
chop $version_number;
($version_tag = $MODNAME."-".$version_number) =~ s/\./_/g;

$CVSCOM = 'cvs';
$COM = qq/$CVSCOM commit -m'$TAO_VERSION' $VERSION $CHANGELOG && $CVSCOM rtag $version_tag $MODNAME && $CVSCOM rtag -F Current $MODNAME/;
print $COM;
system $COM;

unlink @inplace;
